src/renderer/rendered_window.rs

```diff
@@ -455,71 +464,92 @@ impl RenderedWindow {
-                translated_region.offset((
-                    -cols as f32 * font_width as f32,
-                    -rows as f32 * font_height as f32,
-                ));
-
-                let snapshot = self.current_surface.surface.image_snapshot();
-                let canvas = self.current_surface.surface.canvas();
-
-                canvas.save();
-
-                canvas.clip_rect(scrolled_region, None, Some(false));
-                canvas.draw_image_rect(
-                    snapshot,
-                    Some((&scrolled_region, SrcRectConstraint::Fast)),
-                    translated_region,
-                    &grid_renderer.paint,
-                );
-
-                canvas.restore();
+                tracy_zone!("scroll_cmd", 0);
+                if top == 0
+                    && bottom == self.grid_size.height
+                    && left == 0
+                    && right == self.grid_size.width
+                    && cols == 0
+                {
+                    self.actual_top_index += rows as isize;
+                }
             }
             WindowDrawCommand::Clear => {
-                self.current_surface.surface = build_window_surface_with_grid_size(
-                    self.current_surface.surface.canvas(),
-                    grid_renderer,
-                    self.grid_size,
-                );
-
-                self.snapshots.clear();
+                tracy_zone!("clear_cmd", 0);
+                self.actual_top_index = 0;
+                self.scrollback_top_index = 0;
+                self.scrollback_lines
+                    .iter_mut()
+                    .for_each(|line| *line = None);
+                self.reset_scroll();
             }
             WindowDrawCommand::Show => {
+                tracy_zone!("show_cmd", 0);
                 if self.hidden {
                     self.hidden = false;
                     self.position_t = 2.0; // We don't want to animate since the window is becoming visible,
                                            // so we set t to 2.0 to stop animations.
                     self.grid_start_position = self.grid_destination;
+                    self.reset_scroll();
                 }
             }
-            WindowDrawCommand::Hide => self.hidden = true,
-            WindowDrawCommand::Viewport { scroll_delta, .. } => {
-                if scroll_delta.abs() > f64::EPSILON {
-                    let new_snapshot = self.current_surface.snapshot();
-                    self.snapshots.push_back(new_snapshot);
-
-                    if self.snapshots.len() > 5 {
-                        self.snapshots.pop_front();
+            WindowDrawCommand::Hide => {
+                tracy_zone!("hide_cmd", 0);
+                self.hidden = true;
+            }
+            WindowDrawCommand::Viewport { scroll_delta } => {
+                // The scroll delta is unfortunately buggy in the current version of Neovim. For more details see:
```

Comment:
Probably intended to reference to https://github.com/neovim/neovim/issues/23609.